### ISSUE TITLE
Corner case of skipmissing

### ISSUE THREAD

**Issue by bkamins** [JuliaLang/julia#35977]
This is a minor thing, but I guess we do not want to have dispatch ambiguities in Base:
```
julia> sum(skipmissing([missing]))
ERROR: MethodError: reduce_empty(::typeof(Base.add_sum), ::Type{Union{}}) is ambiguous. Candidates:
  reduce_empty(::typeof(Base.add_sum), ::Type{T}) where T<:Union{Int16, Int32, Int8} in Base at reduce.jl:314
  reduce_empty(::typeof(Base.add_sum), ::Type{T}) where T<:Union{UInt16, UInt32, UInt8} in Base at reduce.jl:315
Possible fix, define
  reduce_empty(::typeof(Base.add_sum), ::Type{Union{}})
```

CC @nalimilan

**Comment by tkf** [comment]
#35843 will fix this

**Comment by bkamins** [comment]
It is interesting how things go in parallel in the same time. We are cleaning up reduce code for `GroupedDataFrame` in DataFrames.jl and the same issues pop-up :smile:.

**Comment by tkf** [comment]
closed by #35843
